.github/workflows/format.yml

```diff
@@ -0,0 +1,37 @@
+name: lint
```

Comment:
`pre-commit` hooks already supports to clang-format check. Maybe instead of adding a new runner enabling it from config file (https://github.com/seladb/PcapPlusPlus/blob/3ffed8250fde57733f9895df4461a7096e21aae3/.pre-commit-config.yaml#L17-L22) might be better. People can run `pre-commit run --all-files` to check locally. As a default it search `.clang-format` in workspace directory